src/sharechain/in_memory.rs

```diff
@@ -563,12 +600,22 @@ impl ShareChain for InMemoryShareChain {
         }
 
         for (key, value) in miners_to_shares {
+            // find coinbase extra for wallet address
+            let mut coinbase_extra = convert_coinbase_extra(tribe.clone(), String::new()).unwrap_or_default();
+            if let Ok(miner_wallet_address) = TariAddress::from_str(key.as_str()) {
+                if let Some(coinbase_extra_found) =
+                    self.find_coinbase_extra(bl.levels.iter(), miner_wallet_address).await
```

Comment:
This is O(n^2) and will be slow, I think find a better way of doing this without iterating through all the levels

Reply:
totally makes sense, let me refactor

Reply:
Okay just pushed!
I removed the unneeded iteration, just using the provided coinbase cache and update there the values when adding new blocks to share chain.